src/core/thread/mesh_forwarder.cpp

```diff
@@ -999,6 +1026,18 @@ void MeshForwarder::HandleSentFrame(Mac::TxFrame &aFrame, Error aError)
 
     VerifyOrExit(mEnabled);
 
+#if OPENTHREAD_CONFIG_MAC_DELAY_NEXT_TX_AFTER_FORWARDING_FRAME_TX
+    if (mDelayNextTx && (aError == kErrorNone))
+    {
+        mTxDelayTimer.Start(kTxDelayInterval);
+        LogDebg("Start tx delay timer for %u msec", kTxDelayInterval);
+    }
+    else
+    {
+        mDelayNextTx = false;
+    }
```

Comment:
Is it a good idea to put these logic to `ScheduleTransmissionTask` ?

Reply:
No, I think this logic should be here in `HandleSentFrame` since  we want to add a "silence"/wait time to start after frame tx done and it is tied to whether the frame tx was successful or not. Moving it to `ScheduleTransmissionTask` won't have the behavior we want.

Reply:
> I think this logic should be here in HandleSentFrame since we want the silence/wait window to start after frame tx is done and it is tied to whether the frame tx was successful or not.

Agree. We would need to set the TX delay deadline time if moving this logic to ScheduleTransmissionTask.

The current code SGTM.